src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -442,7 +442,7 @@ public int LastIndexOf(T item, int startIndex, int count, IEqualityComparer<T> e
             }
             else
             {
-                for (int i = startIndex; i >= startIndex - count + 1; i--)
+                for (int i = startIndex; i > startIndex - count; i--)
```

Comment:
What does this change have to do with the perf improvement? If it is just style, it should probably be separated. It may be easier reading, but it takes time to validate that this doesn't introduce a subtle regression so in the interest of increasing the chances of taking the pull request on its perf gains it's best to avoid distractions